packages/iov-keycontrol/src/wallets/slip10wallet.spec.ts

```diff
@@ -70,6 +70,68 @@ describe("Slip10Wallet", () => {
     expect(wallet.label.value).toBeUndefined();
   });
 
+  describe("previewIdentity", () => {
+    it("works for Secp256k1", async () => {
+      // Test data generated by the BIP44 tool of https://iancoleman.io/bip39/#english
+      // and uncompressed pubkeys from https://www.bitaddress.org/ (Don't get confused
+      // by keygen. Just click "Wallet Details" and enter privkey)
+
+      const wallet = Slip10Wallet.fromMnemonicWithCurve(
+        Slip10Curve.Secp256k1,
+        "mushroom faint dumb venture million true skull grab pitch mesh share tortoise",
+      );
+
+      const identity = await wallet.previewIdentity(defaultChain, [
+        Slip10RawIndex.hardened(44),
+        Slip10RawIndex.hardened(0),
+        Slip10RawIndex.hardened(7),
+        Slip10RawIndex.normal(1),
+        Slip10RawIndex.normal(0),
+      ]);
+      expect(identity).toEqual({
+        chainId: defaultChain,
+        pubkey: {
+          algo: Algorithm.Secp256k1,
+          data: fromHex(
+            "0488557bc34cf8229fc40cffe464344e946bf5c46257e820ea1632f3acbeaa723b40a47bc4015a9b5a89c3ba22dd271ea672526d735cc619a709a65d2201bc079f",
+          ) as PublicKeyBytes,
+        },
+      });
+
+      // preview identity not persisted
+      expect(wallet.getIdentities()).toEqual([]);
+    });
+
+    it("works for Ed25519", async () => {
+      // Test 1 from https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0005.md#test-cases
+      //
+      // Stellar public keys can be converted to raw ed25519 pubkeys as follows
+      // $ yarn add stellar-sdk
+      // $ node
+      // > Keypair.fromPublicKey("GDRXE2BQUC3AZNPVFSCEZ76NJ3WWL25FYFK6RGZGIEKWE4SOOHSUJUJ6").rawPublicKey().toString("hex")
```

Comment:
Where does `Keypair` come from?

Reply:
Thanks, done in 6bf5340fea0aefdf17777538521d017a4ff677c8